xbmc/semantic/SemanticIndexService.cpp

```diff
@@ -0,0 +1,1324 @@
+
+  if (estimatedCost > remainingBudget)
+  {
+    CLog::Log(LOGWARNING, "SemanticIndexService: Estimated cost ${:.4f} exceeds budget ${:.2f}",
+              estimatedCost, remainingBudget);
+    if (m_database->GetIndexState(mediaId, mediaType, state))
+    {
+      state.transcriptionStatus = IndexStatus::FAILED;
+      m_database->UpdateIndexState(state);
+    }
+    return false;
+  }
+
+  // Create temp directory for audio
+  std::string tempDir = CSpecialProtocol::TranslatePath("special://temp/semantic_audio/");
+  XFILE::CDirectory::Create(tempDir);
+
+  // Generate output path
+  std::string audioPath = tempDir + StringUtils::Format("audio_{}_{}.mp3", mediaType, mediaId);
+
+  CLog::Log(LOGINFO, "SemanticIndexService: Extracting audio from {} to {}", mediaPath, audioPath);
+
+  // Extract audio
+  if (!extractor.ExtractAudio(mediaPath, audioPath))
+  {
+    CLog::Log(LOGERROR, "SemanticIndexService: Failed to extract audio from {}", mediaPath);
+    if (m_database->GetIndexState(mediaId, mediaType, state))
+    {
+      state.transcriptionStatus = IndexStatus::FAILED;
+      m_database->UpdateIndexState(state);
+    }
+    return false;
+  }
+
+  CLog::Log(LOGINFO, "SemanticIndexService: Audio extracted, starting transcription");
+
+  // Transcribe with callbacks
+  std::vector<SemanticChunk> chunks;
+  bool transcriptionSuccess = false;
+  std::string transcriptionError;
+
+  // Capture context for callbacks
+  int capturedMediaId = mediaId;
+  std::string capturedMediaType = mediaType;
+
+  transcriptionSuccess = provider->Transcribe(
+      audioPath,
+      // Segment callback - called for each transcribed segment
+      [&chunks, capturedMediaId, &capturedMediaType, &audioPath](const TranscriptSegment& segment) {
+        SemanticChunk chunk;
+        chunk.mediaId = capturedMediaId;
+        chunk.mediaType = capturedMediaType;
+        chunk.sourceType = SourceType::TRANSCRIPTION;
+        chunk.sourcePath = audioPath;
+        chunk.startMs = static_cast<int>(segment.startMs);
+        chunk.endMs = static_cast<int>(segment.endMs);
+        chunk.text = segment.text;
+        chunk.language = segment.language;
+        chunk.confidence = segment.confidence;
+        chunks.push_back(chunk);
+
+        CLog::Log(LOGDEBUG, "SemanticIndexService: Transcribed segment [{}-{}ms]: {}",
+                  segment.startMs, segment.endMs,
+                  segment.text.substr(0, 50) + (segment.text.length() > 50 ? "..." : ""));
+      },
+      // Progress callback
+      [this, capturedMediaId, &capturedMediaType](float progress) {
+        SemanticIndexState progressState;
+        if (m_database->GetIndexState(capturedMediaId, capturedMediaType, progressState))
+        {
+          progressState.transcriptionProgress = progress;
+          m_database->UpdateIndexState(progressState);
+        }
+        CLog::Log(LOGDEBUG, "SemanticIndexService: Transcription progress: {:.1f}%", progress * 100);
```

Comment:
Please update to the current log standards using `{}`.

For more information please see the [libfmt documentation](https://fmt.dev/latest/syntax.html)